shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/config/schema/impl/DataSourceGeneratedSchemaConfiguration.java

```diff
@@ -41,14 +41,27 @@
     private final Map<String, DataSource> dataSources;
     
     private final Collection<RuleConfiguration> ruleConfigurations;
+
+    private final Map<String, DataSourceProperties> dataSourceProperties;
     
     public DataSourceGeneratedSchemaConfiguration(final Map<String, DataSourceConfiguration> dataSources, final Collection<RuleConfiguration> ruleConfigurations) {
         this.dataSources = DataSourcePoolCreator.create(createDataSourcePropertiesMap(dataSources));
         this.ruleConfigurations = ruleConfigurations;
+        this.dataSourceProperties = createDataSourcePropertiesMap(dataSources);
     }
     
     private Map<String, DataSourceProperties> createDataSourcePropertiesMap(final Map<String, DataSourceConfiguration> dataSources) {
         return dataSources.entrySet().stream().collect(Collectors.toMap(Entry::getKey,
             entry -> DataSourcePropertiesCreator.create("com.zaxxer.hikari.HikariDataSource", entry.getValue()), (oldValue, currentValue) -> oldValue, LinkedHashMap::new));
     }
+
+    /**
+     * Get data source properties.
+     *
+     * @return data source map
+     */
+    public Map<String, DataSourceProperties> getDataSourceProperties() {
+        return this.dataSourceProperties;
+    }
```

Comment:
Can replace it to lombok's @Getter

Reply:
Fix in https://github.com/apache/shardingsphere/pull/16632